framework/test/gtest/BundleTrackerMethodTest.cpp

```diff
@@ -0,0 +1,480 @@
+
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  bundle.Uninstall();
+  EXPECT_EQ(1, bundleTracker.Size())
+    << "Size should be 1 when 1 bundle entered a tracked state";
+
+  bundleTracker.Close();
+  EXPECT_EQ(0, bundleTracker.Size())
+    << "Size of closed BundleTracker was not 0";
+}
+
+TEST_F(BundleTrackerMethodTest, GetTrackingCountWorksAfterBundleAdded)
+{
+  auto stateMask =
+    BundleTracker<>::CreateStateMask(Bundle::State::STATE_ACTIVE);
+  BundleTracker<> bundleTracker(context, stateMask);
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+
+  int preCount = bundleTracker.GetTrackingCount();
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  bundle.Start(); // bundle: installed->resolved->starting->ACTIVE
+  int postCount = bundleTracker.GetTrackingCount();
+
+  EXPECT_EQ(1, postCount - preCount)
+    << "Tracking count didn't increment by 1 after a bundle was added";
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest, GetTrackingCountWorksAfterBundleModified)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  ASSERT_EQ(Bundle::State::STATE_INSTALLED, bundle.GetState())
+    << "Test bundle failed to install";
+
+  int preCount = bundleTracker.GetTrackingCount();
+  bundle.Start(); // bundle: INSTALLED->RESOLVED->STARTING->ACTIVE
+  ASSERT_EQ(Bundle::State::STATE_ACTIVE, bundle.GetState())
+    << "Test bundle failed to start";
+  int postCount = bundleTracker.GetTrackingCount();
+
+  EXPECT_EQ(3, postCount - preCount)
+    << "Tracking count didn't increment by 3 after a bundle was modified 3 "
+       "times";
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest,
+       GetTrackingCountWorksAfterBundleRemovedByStateChange)
+{
+  auto stateMask =
+    BundleTracker<>::CreateStateMask(Bundle::State::STATE_INSTALLED);
+  BundleTracker<> bundleTracker(context, stateMask);
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  ASSERT_EQ(Bundle::State::STATE_INSTALLED, bundle.GetState())
+    << "Test bundle failed to install";
+
+  int preCount = bundleTracker.GetTrackingCount();
+  bundle.Start(); // bundle: INSTALLED->resolved->starting->active
+  ASSERT_EQ(Bundle::State::STATE_ACTIVE, bundle.GetState())
+    << "Test bundle failed to start";
+  int postCount = bundleTracker.GetTrackingCount();
+
+  EXPECT_EQ(1, postCount - preCount)
+    << "Tracking count didn't increment by 1 after a bundle was removed";
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest, GetTrackingCountWorksAfterRemoveMethod)
```

Comment:
minor; rename testpoint name - remove "Works" to be more consistent with other test point names

Reply:
Done.